calyx-frontend/src/ast.rs

```diff
@@ -63,10 +63,22 @@ pub struct ComponentDef {
     pub attributes: Attributes,
     /// True iff this is a combinational component
     pub is_comb: bool,
+    /// True iff this is a static component
+    pub is_static: bool,
+    /// (Optional) latency of component, if it is static
+    /// NOTE: latency can still be None even if is_static is true. That
+    /// just means the compiler will infer the latency
+    pub latency: Option<NonZeroU64>,
```

Comment:
Hm, this is an interesting design choice. I was thinking that we only have the `latency` field and nothing else. I'm not sure what having `is_static` in addition to that accomplishes. Is the idea that there are three kinds of components:
1. Static components with specified latency
2. Static components without specified latency
3. Dynamic components

If that is the case, then (2) & (3) should probably be same for the following reason: Using the `static` keyword on a component is declaring a calling convention–it says that it is not safe to call this component using a dynamic invoke. In contrast, a component that merely *happens* to use only `static` groups and control is not declaring its interface as `static` and the compiler is free to choose whether it is promoted to a static component or compiled into a dynamic one.